### PR TITLE
Catch more connection exceptions and keep retrying

### PR DESCRIPTION
**Pull request by jeremych1000**
Fixes https://github.com/Hari-Nagarajan/nvidia-bot/issues/72

image

### CODE REVIEW

**Comment by jeremych1000** [comment]
I'm using `black` as well, so not sure why some of the formatting changes

File: stores/nvidia.py [hunk 1]
```diff
@@ -456,13 +461,16 @@ def add_to_cart(self, product_id):
...
                 return False
-        except Exception as ex:
-            log.warning(str(ex))
-            log.warning("The connection has been reset.")
+        except (
+            ConnectionResetError,
+            requests.exceptions.ConnectionError,
+            requests.exceptions.Timeout,  # covers ConnectTimeout + ReadTimeout
+        ) as ex:
+            log.warning(f"The connection experienced a problem: {ex}.")
```
Comment (Hari-Nagarajan) [inline stores/nvidia.py:473]: base Exception already catches all of these I believe.

Comment (jeremych1000) [reply stores/nvidia.py:473]: Yup, it does :)

**Comment by jeremych1000** [comment]
Closed as [this commit](https://github.com/Hari-Nagarajan/nvidia-bot/commit/e35914af92d08e0e3f6421b2736c95e0ea94131b) catches it anyway

**Pull request closed without merging**